internal/node/reverse_proxy/server.go

```diff
@@ -37,33 +25,49 @@ func NewProxyServer(logger *zerolog.Logger, config *config.NodeConfig) *ProxySer
 	return &ProxyServer{
 		logger: logger,
 		RuleSet: &RuleSet{
-			rules:        make(map[string]RuleHandler),
+			rules:        make(map[string]*node.ReverseProxyRule),
 			baseFilePath: config.WebBundlePath(),
 		},
 		nodeConfig: config,
 	}
 }
 
 func (s *ProxyServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	var bestMatch RuleHandler = nil
+	var bestMatch *node.ReverseProxyRule = nil
 	// Find the matching rule with the highest "rank", aka the most slashes '/' in the URL path.
 	highestRank := -1
 	for _, rule := range s.RuleSet.rules {
-		if rule.Match(r.URL) {
-			if rule.Rank() > highestRank {
-				bestMatch = rule
-				highestRank = rule.Rank()
+		if rule != nil {
+			if matchRule(r.URL, rule) {
+				rank := rankMatch(r.URL, rule)
```

Comment:
when can we get rid of the ranking logic? i thought go had some default stuff in the http package that could do this

Reply:
Do you know where that is? Overall, I think we'll probably always need something like this, especially if there's some dynamic matching fro routes in the future.